### PR TITLE
travis matrix testing

### PR DESCRIPTION
**Pull request by lynxis**
improves the speed and clean up the code.

### CODE REVIEW

**Comment by lynxis** [comment]
@mitar when the checks are green, this is ready for merge.

**Comment by mitar** [comment]
It is not green. :-(

**Comment by mitar** [comment]
> Dummy Client failed to ping server

This looks like flaky test. :-(

**Comment by lynxis** [comment]
On Friday, May 13, 2016 6:27:43 PM CEST, Mitar <notifications@github.com> 
wrote:
>> Dummy Client failed to ping server
> 
> This looks like flaky test. :-(

the dummy client serves two functions.
- increase the usage count of one broker
- tests if the client can connect.

i'm not sure why its failing, but there is an exception on the broker side


-- 
Sent using Dekko from my Ubuntu device

**Comment by mitar** [comment]
You are right. So this did catch a bug.

File: .travis.yml [hunk 1]
```diff
@@ -15,4 +15,12 @@ install:
...
   - sudo /usr/bin/python3 ./tests/tunneldigger.py --setup
 script:
-  - sudo sh ./tests/travis.sh
+  - sudo -E sh ./tests/travis.sh
+
+env:
+  matrix:
+    - SELECT=nose OLD_REV=HEAD
+    - SELECT=nose OLD_REV=c638231efca6b3a6e1c675ac0834a3e851ad1bdc
+    - SELECT=nose OLD_REV=4e4f13cdc630c46909d47441093a5bdaffa0d67f
```
Comment (mitar) [inline .travis.yml]: Didn't you have one more revision to test against?

Comment (lynxis) [inline .travis.yml]: yes, I removed one, because of the long time of the test, but we can reenable it, because it shouldn't increase the test as much.

**Comment by lynxis** [comment]
```
[INFO/tunneldigger.hooks] Running hook 'session.up' via script '/testing/hook_server/setup_interface.sh 100 1 l2tp1001 1446 172.16.16.2 54733 20100 oobar'.
[ERROR/tunneldigger.hooks] DEBUG  hook 'session.up' via script '/testing/hook_server/setup_interface.sh (100, 1, 'l2tp1001', 1446, '172.16.16.2', 54733, 20100, 'oobar\x00\x00\x00\x02')'.
[ERROR/tunneldigger.hooks] DEBUG:      'session.up' '/testing/hook_server/setup_interface.sh' '(100, 1, 'l2tp1001', 1446, '172.16.16.2', 54733, 20100, 'oobar\x00\x00\x00\x02')'
[ERROR/tunneldigger.broker] Unhandled exception while creating tunnel 100:
[ERROR/tunneldigger.broker] Traceback (most recent call last):
  File "broker/broker.py", line 74, in create_tunnel
    tunnel.setup_tunnel()
  File "broker/tunnel.py", line 186, in setup_tunnel
    self.uuid,
  File "broker/hooks.py", line 148, in run_hook
    process = HookProcess(name, script, args)
  File "broker/hooks.py", line 32, in __init__
    stderr=subprocess.PIPE,
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
    raise child_exception
TypeError: execv() arg 2 must contain only strings
```

File: .travis.yml (hunk 1 above)
Comment (mitar) [inline .travis.yml]: Yes, add it. :-)

**Comment by lynxis** [comment]
@mitar there are two errors. One I took a bad version of the NONUSAGE_REV. Second the client connected to it, even it shouldn't because it wasn't using the usage broker selector at all.

**Comment by lynxis** [comment]
The usage is broken for real, not only the test. #43 will fix that late.

**Comment by lynxis** [comment]
@mitar this is now ready-for-merge. the usage is still broken.

**Pull request merged**